src/services/RetroGenerationService.ts

```diff
@@ -46,15 +46,18 @@ class RetroGenerationService {
                     const testType: any = testResult.testTypes;
                     const additionalTestTypeNotes = testType.prohibitionIssued ? 'Prohibition was issued' : 'Prohibition was not issued'
 
-                    //TODO Refactor
-                    let prsString = 'PRS';
-                    try { if (testType.defects[i].prs === undefined) prsString = 'PRS';} catch (e) {
-                        prsString = 'NOT PRS';
+                    let prsString = ''
+                    if(testType.defects[i]) {
+                        if (!testType.defects[i].prs) {
```

Comment:
we might want to test this on a feature branch first, with both PRS and without PRS scenarios. Have a feeling this might not work. Can we validate this via a unit test pls?

Reply:
tested on feature branch together with frontend app. Looks ok with prs=null. can't find any way to send prs undefined from app though.